v3/src/data-model/data-set.ts

```diff
@@ -433,9 +445,11 @@ export const DataSet = types.model("DataSet", {
       setAttributeName(attributeID: string, name: string) {
```

Comment:
What if this were changed so that the `setAttributeName()` function took either a `string` or a function which returned a string, i.e.
```
  setAttributeName(attributeID: string, name: string | (() => string)) {
    ...
```
and then clients that want to enforce uniqueness can do so by passing something like:
```
  data.setAttributeName(attrId, () => uniqueName(name, ...))
```
and pass an appropriate `isValid()` function?